### PR TITLE
Hp/sign up mutation

### PR DESCRIPTION
**Pull request by hernan-dp**
Sign Up Mutation with it's resolver and controller

### CODE REVIEW

File: src/controllers/user-controller.js [hunk 1]
```diff
@@ -15,3 +17,19 @@ export const findAllUsers = (obj, args, { models }) => {
 export const findUserById = (_, { id }, { models }) => {
   return models.user.findByPk(id)
 }
+
+export const signUp = async (_, data, { models }) => {
+  const user = await models.user.findOne({ where: { username: input.username }}) 
+  if (!user){
+    user = await models.user.createUser({ data })
```
Comment (abelosorio) [inline src/controllers/user-controller.js:24]: You have declared user as a constant. You can't override its value.

File: src/controllers/user-controller.js [hunk 2]
```diff
@@ -15,3 +17,19 @@ export const findAllUsers = (obj, args, { models }) => {
...
+    const secret = process.env.JWT_SECRET;
+    const token = jwt.sign({ sub: user.id }, secret, { exp: '10d' } );
+    return {
+      token,
+      user,
+    }
+  }
+  else {
+    throw new Error('Usuario ya existente');
```
Comment (abelosorio) [inline src/controllers/user-controller.js]: Fail loud, fail early.

This should be much clearer if you do it in the first line:
```javascript
if (user) throw new Error(...)

// the rest of your code here
```

File: src/controllers/user-controller.js [hunk 3]
```diff
@@ -15,3 +17,19 @@ export const findAllUsers = (obj, args, { models }) => {
 export const findUserById = (_, { id }, { models }) => {
   return models.user.findByPk(id)
 }
+
+export const signUp = async (_, data, { models }) => {
+  const user = await models.user.findOne({ where: { username: input.username }}) 
+  if (!user){
+    user = await models.user.createUser({ data })
+    const secret = process.env.JWT_SECRET;
+    const token = jwt.sign({ sub: user.id }, secret, { exp: '10d' } );
```
Comment (abelosorio) [inline src/controllers/user-controller.js]: 10 days for an access token is wide too much time. Take a look at this video: https://www.youtube.com/watch?v=SLc3cTlypwM.

**Review by abelosorio — CHANGES_REQUESTED** [review]

File: src/controllers/user-controller.js (hunk 3 above)
Comment (hernan-dp) [reply src/controllers/user-controller.js]: Hi Abel, in the requirements passed in this module says it's 10 days https://github.com/WeDevelop-ARG/full-stack-beginner-program/blob/master/content/block-4/jwt-details.md.
Do you think less time is more appropriate?

**Review by abelosorio — APPROVED** [review]
Great!

**Review by nahudalla — APPROVED** [review]
@hernan-dp please resolve the conflicts and feel free to merge the PR once done.

**Comment by mpautasso** [comment]
@hernan-dp Do you have some conflicts, could you rebase with `develop` branch, please?. Thanks

**Pull request merged**